ARCHITECTURE.adoc

```diff
@@ -0,0 +1,239 @@
+work through those in conceptual order.
+
+== Plutus Core
+
+Plutus Core is the language that actually goes on the blockchain. Consequently
+this is the absolute core of the codebase, and everything depends on it.
+
+=== `language-plutus-core`
+
+This package implements the Plutus Core language.
+
+This includes:
+
+- Parser
+- Typechecker
+- Prettyprinter
+- Simple evaluator (see `plutus-core-evaluator` for fancier evaluators)
+
+=== `plutus-core-intepreter`
+
+This package contains the production evaluator that we use, along with an experimental
+lazy evaluator.
+
+=== `plutus-exe`
+
+This package defines a command-line executable that used to typecheck or
+evaluate Plutus Core programs.
+
+== Plutus Tx
+
+Plutus Tx is how we refer to the subset of Haskell which we compile into Plutus
+Core. This is how users actually write Plutus contracts: they write Haskell
+programs, part of which is compiled into Plutus Core. The rest of the program
+can then use this compiled code when submitting transactions.
+
+=== `plutus-ir`
+
+Plutus IR is a higher-level language that sits between Plutus Tx and Plutus
+Core in the compilation pipeline. This package implements the compiler, which
+compiles Plutus IR into Plutus Core.
+
+=== `plutus-tx`
+
+This package provides several things:
+
+- The Plutus Tx compiler, which compiles GHC Core into
+Plutus IR.
+- A GHC Core plugin, which is the mechanism by which people use
+the compiler.
+- A couple of special typeclasses which aid with the interface
+between Haskell and Plutus Tx, and has some Template Haskell support for
+generating instances.
+- It provides a partial replacement Prelude, since many parts of the
+normal Haskell Prelude cannot be used with Plutus Tx.
+
+== Ledger
+
+To write actual Plutus contracts we need a model of the ledger that we're
+running on. There are two reasons for this:
+
+- The on-chain code needs to work with the data: validators are given information
+  about the current transaction being validated.
+- The off-chain code needs to produce transactions to submit.
+- We want to write tests that simulate the "full" behaviour of contracts, i.e.
+  across time, in a multi-agent scenario.
+
+=== `plutus-wallet-api`
+
+This package should probably be split in two!
+
+The `ledger` sublibrary defines our model of an Extended UTXO ledger, including:
+- The types that
+
+The rest of the package defines the
+
+=== `plutus-emulator`
+
+This package defines the chain emulator, which is used for tests, and to back
+the simulations in the Plutus Playground.
+
+== Contract modelling
+
+We want to help people write "contracts". When we say that, we mean both:
+. A high-level applications that exposes a limited API and
+is suitable for running inside something like a wallet smart contract backend.
+. An implementation of a conceptual "contract" (e.g. a crowdfunding contract),
+including both the on-chain code and the eventual contract application.
+
```

Comment:
I think the markdown's wrong here.  It renders as "When we say that, we mean both: . A high-level applications that exposes a limited API and is suitable for running inside something like a wallet smart contract backend. . An implementation of a conceptual "contract" (e.g. a crowdfunding contract), including both the on-chain code and the eventual contract application."  Does it need some indentation at the start of lines?

Reply:
This is asciidoc, and it needs a blank line before the start of a list, which I always forget.